### ISSUE TITLE
[method-signature-style] respect getter signature

### ISSUE THREAD

**Issue by AkatQuas** [typescript-eslint/typescript-eslint#4762]
- [x] I have tried restarting my IDE and the issue persists.
- [ ] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/method-signature-style": ["error", "property"],
  }
}
```

```TS
type IService = {
  get age(): number;
}
```

**Expected Result**

`age` is still a `getter` property.

```TS
// don't auto fix
type IService = {
  get age(): number;
}
```

**Actual Result**

Now the `age` is a function.

```TS
// however, auto fix happends
type IService = {
  age: () => number;
}
```

**Additional Info**

None.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.33.0` |
| `@typescript-eslint/parser`        | `4.33.0` |
| `TypeScript`                       | `4.4.4` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `14.19.1` |

**Comment by bradzacher** [comment]
@AkatQuas please make sure you fill out the entire template next time without deleting any sections. It is important that you give us all information.

**Comment by AkatQuas** [comment]
Sorry about that. I have added some version information. 

As far as I concerned, this rule should respect the getter signature.

**Comment by bradzacher** [comment]
Out of curiosity - what's your usecase for a getter signature? Why not just use a property signature?

**Comment by AkatQuas** [comment]
Here is my usecase.

Supposing we have some code like this

```ts
class A implements IA {
    private _age = 2
    get age() {
        return this._age;
    }
}

type IA = {
    /** this is age */
    get age(): number;
}
```

When using getter signature, the TS annotation works well.

<img width="285" alt="image" src="https://user-images.githubusercontent.com/24731539/160962807-772b894b-0dfd-40ca-9d19-f1d6c459d463.png">

However, TS annotation fails at property signature.

```ts
class A implements IA {
    get age() {
        return this._age;
    }

    private _age = 2;
}

type IA = {
    /** this is age */
    age: number;
}
```

<img width="289" alt="image" src="https://user-images.githubusercontent.com/24731539/160962928-29356bc9-987e-4c9e-89e2-e560d3c7767e.png">

**Comment by bradzacher** [comment]
That's really interesting! Thanks. 
I'm surprised that TS doesn't inherit the JSDoc for interface properties.
